packages/nuxt/src/module.ts

```diff
@@ -68,13 +82,12 @@ export default defineNuxtModule<ModuleOptions>({
       const registerOnyxLocales: NuxtI18nModuleHooks["i18n:registerModule"] = (register) => {
         register({
           langDir: resolve("./runtime/locales"),
-          locales: [
-            // we need to use .js files instead of .ts because the .ts files would be compiled to .js in the build step, so
-            // when projects use this nuxt module, .ts files will throw a "can not find file" error
-            { code: "de-DE", file: "de-DE.js" },
-            { code: "en-US", file: "en-US.js" },
-            { code: "ko-KR", file: "ko-KR.js" },
-          ],
+          // we need to use .js files instead of .ts because the .ts files would be compiled to .js in the build step, so
+          // when projects use this nuxt module, .ts files will throw a "can not find file" error
+          locales: Object.entries(options.i18n?.registerLocales ?? {}).map(([code, language]) => ({
```

Comment:
Can't we just check which locales the user has defined in nuxt i18n and only register onyx locales then?

Reply:
Yes but actually no. :D
We could do that but as the locales can be defined in a lot of different ways in different places this would be really fragile and error prone. It would also not work if the user uses different locale codes than BCP-47.
And most important: This is only intended as a temporary fix to get the module working in the first place. Later nuxt-i18n should handle all the merging and mapping for the locales.